cedar-policy-core/src/parser/loc.rs

```diff
@@ -74,6 +82,101 @@ impl From<&Loc> for miette::SourceSpan {
     }
 }
 
+/// Trait to define conversions to `Option<&Loc>`
+pub trait AsLocRef {
+    /// Automatic conversion to `Option<&Loc>`
+    fn as_loc_ref(&self) -> Option<&Loc>;
+}
+
+impl AsLocRef for Option<Loc> {
+    #[inline]
+    fn as_loc_ref(&self) -> Option<&Loc> {
+        self.as_ref()
+    }
+}
+
+impl AsLocRef for Option<Box<Loc>> {
+    #[inline]
+    fn as_loc_ref(&self) -> Option<&Loc> {
+        self.as_deref()
+    }
+}
+
+/// Trait to define conversions into `MaybeLoc`
+pub trait IntoMaybeLoc {
+    /// Automatic conversion to `MaybeLoc`
+    fn into_maybe_loc(self) -> MaybeLoc;
+}
+
+impl IntoMaybeLoc for Loc {
+    #[inline]
+    fn into_maybe_loc(self) -> MaybeLoc {
+        #[cfg(not(feature = "fast-parsing"))]
+        {
+            Some(self)
+        }
+        #[cfg(feature = "fast-parsing")]
+        {
+            Some(Box::new(self))
+        }
+    }
+}
+
+impl IntoMaybeLoc for Option<Loc> {
+    #[inline]
+    fn into_maybe_loc(self) -> MaybeLoc {
+        #[cfg(not(feature = "fast-parsing"))]
+        {
+            self
+        }
+        #[cfg(feature = "fast-parsing")]
+        {
+            self.map(|loc| Box::new(loc))
+        }
+    }
+}
+
+impl IntoMaybeLoc for Option<&Loc> {
+    #[inline]
+    fn into_maybe_loc(self) -> MaybeLoc {
+        #[cfg(not(feature = "fast-parsing"))]
+        {
+            self.cloned()
+        }
+        #[cfg(feature = "fast-parsing")]
+        {
+            self.map(|loc| Box::new(loc.clone()))
+        }
+    }
+}
+
+impl IntoMaybeLoc for Option<Box<Loc>> {
+    #[inline]
+    fn into_maybe_loc(self) -> MaybeLoc {
+        #[cfg(not(feature = "fast-parsing"))]
+        {
+            self.map(|loc| *loc)
+        }
+        #[cfg(feature = "fast-parsing")]
+        {
+            self
+        }
+    }
+}
+
+/// Trait to define clones for `MaybeLoc`
+pub trait CloneMaybeLoc {
+    /// Automatic clone to `MaybeLoc`
+    fn clone_maybe_loc(&self) -> MaybeLoc;
+}
+
+impl CloneMaybeLoc for MaybeLoc {
+    #[inline]
+    fn clone_maybe_loc(&self) -> MaybeLoc {
+        self.clone()
+    }
+}
```

Comment:
why not just use the `Clone` trait for this?

Reply:
Right, I think it's because I started with a trait with multiple `impl`s (for `Option<Loc>`, `Option<Box<Loc>>`...), then changed it to `MaybeLoc` and didn't realize it wasn't needed anymore. Anyway, it's been removed.